File: web/src/views/private/Resultado/validations/resultado.ts
import * as Yup from "yup";

export const resultadoSchema = Yup.object({
  fechaIngreso: Yup.date().required("Fecha de ingreso es requerida"),
  fechaResultado: Yup.date().required("Fecha de resultado es requerida"),
  fechaEntregado: Yup.date().required("Fecha de entrega es requerida"),
  resultadoPaciente: Yup.string().required(
    "Resultado del paciente es requerido"
  ),
  metodo: Yup.string().required("Método es requerido"),
  valorResultado: Yup.string().required("Valor del resultado es requerido"),
  valorReferencia: Yup.string().required("Valor de referencia es requerido"),
  observacion: Yup.string().nullable(),
  idCartilla: Yup.number().required("Cartilla es requerida"),
  idLaboratorio: Yup.number().required("Laboratorio es requerido"),
});

export interface ResultadoForm extends Yup.InferType<typeof resultadoSchema> {}

File: server/Models/Resultado.cs
namespace server.Models
{
    public class Resultado : Base
    {
        public required DateOnly FechaIngreso { get; set; }
        public required DateOnly FechaResultado { get; set; }
        public required DateOnly FechaEntregado { get; set; }
        public required string ResultadoPaciente { get; set; }
        public required string Metodo { get; set; }
        public required string ValorResultado { get; set; }
        public required string ValorReferencia { get; set; }
        public required string? Observacion { get; set; }
        public int IdCartilla { get; set; }
        public int IdLaboratorio { get; set; }
        public required bool Envio { get; set; } 
        public required Cartilla Cartilla { get; set; }
        public required Laboratorio Laboratorio { get; set; }
    }
}